src/components/fx/hover.js

```diff
@@ -1518,7 +1518,7 @@ function hoverAvoidOverlaps(hoverLabels, axKey, fullLayout) {
             posref: d.posref,
             size: d.by * (axIsX ? YFACTOR : 1) / 2,
             pmin: 0,
-            pmax: (axIsX ? fullLayout.width : fullLayout.height)
+            pmax: (axIsX ? fullLayout.width : fullLayout.height) - ax._mainLinePosition
```

Comment:
Here one should also take into account the `side` of the axis (i.e. *top* & *left* / *right*?).
Right now for an `xaxis` with `side` set to *top*, the position is not correct: [demo](https://codepen.io/MojtabaSamimi/pen/gOjogee).

Reply:
Yes, if the general approach of modifying the boundaries here is fine, I'll fix this and add tests for all cases (axis and sides).

Reply:
I like it! In addition to x/y and side let's only make this change if we're using one of the "compare" modes (`hovermode: 'x'|'y'`) - there can be cases with multiple labels even when `hovermode: 'closest'`, for example box plots.
<img width="664" alt="Screenshot 2023-01-23 at 17 12 12" src="https://user-images.githubusercontent.com/2678795/214161543-5d4bdfc8-a909-404b-ad0b-55e13e32f64b.png">